caluma/extensions/visibilities.py

```diff
@@ -28,19 +28,38 @@ def filter_queryset_for_all(self, node, queryset, info):  # pragma: no cover
     @filter_queryset_for(Answer)
     def filter_queryset_for_answer(self, node, queryset, info):
         return queryset.filter(
-            document__family__in=self.filter_queryset_for_document(
-                None, form_models.Document.objects, info
+            Q(
+                document__family__in=self.filter_queryset_for_document(
+                    None, form_models.Document.objects, info
+                )
+            )
+            | (
+                Q(
+                    question__slug__in=[
+                        question
+                        for sublist in settings.REVISION_QUESTIONS.values()
+                        for question in sublist
+                    ]
+                )
+                & Q(
+                    document__family__in=self.filter_queryset_for_document(
+                        None,
+                        form_models.Document.objects,
+                        info,
+                        list(settings.REVISION_QUESTIONS.keys()),
+                    )
+                )
             )
         )
 
     @filter_queryset_for(Document)
-    def filter_queryset_for_document(self, node, queryset, info):
+    def filter_queryset_for_document(
+        self, node, queryset, info, tasks=settings.APPLICANT_TASK_SLUGS
+    ):
         user = info.context.user
         case_ids = get_cases_for_user(user)
 
-        applicant_work_item = Q(
-            family__work_item__task__slug__in=settings.APPLICANT_TASK_SLUGS
-        ) | (
+        applicant_work_item = Q(family__work_item__task__slug__in=tasks) | (
```

Comment:
I think we should change the wording here, as `applicant_work_item` can be a workitem that does not belong to the applicant. Let's just call it `work_item`.

Reply:
Renamed